manifests/agent/config.pp

```diff
@@ -15,24 +11,28 @@
     value   => $puppet::agent::server,
   }
 
+  if $puppet::agent::ca_server {
```

Comment:
Doesn't work, because the value of `$puppet::agent::ca_server' is string, not boolean.